lib/foreman_inventory_upload/async/delayed_start.rb

```diff
@@ -0,0 +1,51 @@
+module ForemanInventoryUpload
+  module Async
+    module DelayedStart
+      extend ActiveSupport::Concern
+
+      START_WINDOW = 3.hours.seconds
+
+      def after_delay(delay = nil, logger: nil, &block)
+        logger ||= self.logger if respond_to? :logger
+        delay ||= ForemanRhCloud.requests_delay || Random.new.rand(START_WINDOW)
+        delay = delay.to_i
+
+        logger&.debug("planning a delay for #{delay} seconds before the rest of the execution")
+
+        sequence do
+          plan_action(ForemanInventoryUpload::Async::DelayAction, delay)
+          concurrence(&block)
+        end
+      end
+
+      def humanized_name
+        _('Wait and %s' % super)
+      end
+    end
+
+    class DelayAction < ::Actions::EntryAction
+      Wake = Algebrick.atom
+
+      def plan(delay)
+        plan_self(delay: delay)
+      end
+
+      def run(event = nil)
+        case event
+        when nil
+          action_logger.debug("Going to sleep for #{delay} seconds")
+          plan_event(Wake, delay)
+          suspend
+        when Wake
+          action_logger.debug('Waking up')
+        else
+          super
+        end
+      end
+
+      def delay
+        input[:delay].to_i
+      end
```

Comment:
`#delay` is already defined on `EntryAction` and is used the the action itself is used as a root plan action and it is being scheduled for the future. Is is most likely not an issue in this case, but still.